### DISCUSSION TITLE
表格识别的几个问题，希望有负责的这部分功能的大佬可以帮忙解答！！！

### DISCUSSION THREAD

**Discussion by liuzhipengchd** [PaddlePaddle/PaddleOCR#12611] [Q&A] [answered]
有几个问题想请教大佬：
1、现在表格的训练配置文件有： SLANet.yml ，SLANet_ch.yml，SLANet_lcnetv2.yml，table_master.yml，哪个版本的效果最好？

2、我现在表格有些表格是无边框的，这种没法识别到表格，应该怎么处理。（ 是不是1、首先要训练版面识别模型，加入无边框的表格的数据，2、然后在对无边框的表格进行标注，训练表格识别模型），如果这样做，会不会对本应该是纯文本的部分，也当作无边框的表格进行识别了？

3、有些图片是很长的大图片，这种识别的效果很差，有什么解决办法？

**Comment by liuzhipengchd** [comment]
4、补充 问题。。我发现wps中图片转成表格的准确率很高，有大佬可以分析下，用的模型是不是一样？如果想训练出wps的那种效果，应该怎么优化。

**Reply by weiwuxian1998** [reply]
对于无线表，根据我个人的实践，模型对x轴的边界识别不准确，可以根据ocr结果进行预划线，转成半有线表，能够辅助模型判断边界，可以提升一些效果

**Comment by crackso** [comment, marked as answer]
1，SLANet是百度的衍生算法，参考平安，table_master是平安的算法，ch的是中文表格；lcnetv2是SLANet的第2个版本，更改了多stage训练及Backbone的算法，这块是新的，还没正式发布吧。不知道效果如何。SLANet和table_master看你需求；

2，表格是表格模型，先别扯到版面识别去，在整体pipeline中，需要用到好几个模型。表格是其中一环。表格模型上无线框和有线框大都数看到的模型是分开的，但其实训练的时候也不影响，混一起也没事，要注意数据集的平衡。识别上，无线框只要你标注好了，识别也会按格子出现；

3，很长的图片在于训练的时候都是要缩放的，越长的特征越不明显，可以加大数据量或者，调整训练缩放参数；

4，人家的模型训练的好，肯定经过了大量的数据的，至于模型是否一样，不得而知。但可以知道的是，目前开源表格模型SLANet和table_master都是可以训练出好模型的，一般情况100+表格模板的训练都是没有问题。如果要训练出一个通用的模型还是比较困难的，数据集和算力都是个大问题；

**Reply by FryBoom** [reply]
直接使用官方的ch_ppstructure_mobile_v2.0_SLANet为什么表格识别模型这么差？

**Comment by SWHL** [comment]
可以参考[TableStructureRec](https://github.com/RapidAI/TableStructureRec)
